### ISSUE TITLE
Message display: line breaks before closing quotation marks

### ISSUE THREAD

**Issue by sebidotorg** [nextcloud/talk-ios#389]
When using the built-in keyboard (or a hardware keyboard) in iOS/iPadOS, the user can set “Smart Punctuation” in the typing options. This will replace the character " by “pretty” or typographically correct quotation marks, according to the language settings. Examples:
“English text”
„German Text“
« French text »

The display code will not treat (at least some of) these characters correctly, often inserting a line break immediately before the closing quotation mark, when it needs to break after it, or break before the preceding word, to keep the last word and the closing quotation mark together (or to prevent line breaks between the quotation marks and a single quoted word). I have seen this happen with German text specifically, I did not test the other languages.

**Comment by Ivansss** [comment]
Hi @sebidotorg :)
Could you share some example when this happen?
I can't reproduce the issue.

![typing](https://user-images.githubusercontent.com/4638605/97430136-b914f500-1918-11eb-9be8-7e5f040e0f9e.gif)

**Comment by sebidotorg** [comment]
Hi,

you just need longer texts, which actually go over more than one line, and which have the quotation mark near the line break. Of course it won‘t happen as long as there does not need to be a line break.

**Comment by sebidotorg** [comment]
Here is an example. There is actually no space between the end of the sentence at „krank.“ and the closing quotation mark, I checked twice:

![3D9E31F8-7EEC-4DA6-8C7D-997359CD8CA9](https://user-images.githubusercontent.com/73582066/97444949-fdaa8b80-192c-11eb-8548-0024adf71d27.jpeg)

(Interestingly, when I use English quotation marks, even Apple Mail will do it wrong and break the line after the *opening* quotation mark in the second sentence of this mail. If I am quoting it with German quotation marks, as I did now, Apple Mail correctly breaks the line just before the „krank.“)

Best regards,
Sebi

**Comment by Ivansss** [comment]
Oh I get it now! :D Thanks for sharing the example!
Unfortunately this seems to be an iOS issue and I guess there is nothing we can do to fix it.

**Comment by sebidotorg** [comment]
If this was an iOS issue, it should also happen in apps like Notes. It does not, I tried. It only happens while typing in Apple Mail (for *opening* English quotation marks, not for the closing ones, and not for German quotation marks at all), and in Nextcloud Talk. There I only get it to happen for the *closing* German quotation marks, not the opening ones, and it does not happen while I type, it only happens when the message is displayed after it has been sent. So this really does not look like an iOS issue to me.

**Comment by sebidotorg** [comment]
Sorry, I have to correct that: it does also happen while typing in Nextcloud Talk, when there is a full stop before the closing quotation mark. I did not check for that earlier, but unlike in English, the punctuation for direct speech has to be inside the quotation marks in German.

I checked Apple Notes, and even with the full stop before the closing quotation mark, the line break is always done in a way that keeps the closing quotation mark together with the end of the quoted sentence.
